ocaml/xapi/repository_helpers.ml

```diff
@@ -1076,3 +1076,24 @@ let prune_updateinfo_for_livepatches livepatches updateinfo =
     List.filter (fun x -> LivePatchSet.mem x livepatches) updateinfo.livepatches
   in
   {updateinfo with livepatches= lps}
+
+let do_with_device_models ~__context ?(vm_with_recommended_guidances = false)
```

Comment:
The meaning of `vm_with_recommended_guidances` given the type `bool` is not obvious. What does this flag represent?

Reply:
If "vm_with_recommended_guidances " is true, "do_with_device_models" will execute "f" with only vm with recommended guidances,
If "vm_with_recommended_guidances " is false, "do_with_device_models" will execute "f" with all vms no matter if a vm has recommended guidances or not.

Reply:
So this acts as a filter. I would suggest to use `only_vm_with_guidances` or `limit_to_vm_with_guidances`.

Reply:
As now "vm" obj has both "recommended_guidances" and "pending_guidances", renamed it to "only_vm_with_recommended_guidances"